mmv1/third_party/terraform/acctest/tgc_utils.go

```diff
@@ -67,7 +69,11 @@ func CollectAllTgcMetadata(tgcPayload TgcMetadataPayload) resource.TestCheckFunc
 			}
 
 			// Resolve the CAI asset name
-			if apiServiceName, ok := ApiServiceNames[metadata.ResourceType]; ok {
+			apiServiceName := GetAPIServiceNameForResource(metadata.ResourceType)
+			if apiServiceName == "unknown" {
+				log.Printf("[DEBUG]TGC Terraform error: unknown resource type %s", metadata.ResourceType)
+				metadata.CaiAssetName = "unknown"
+			} else {
```

Comment:
Does it need to check if apiServiceName is not `failed_to_populate_metadata_cache`?

Reply:
added